Amplify/Categories/Storage/StorageCategory+ClientBehavior.swift

```diff
@@ -95,4 +95,9 @@ extension StorageCategory: StorageCategoryBehavior {
                        options: StorageRemoveRequest.Options? = nil) async throws -> String {
         try await plugin.remove(key: key, options: options)
     }
+
+    @discardableResult
+    public func list(options: StorageListOperation.Request.Options?) async throws -> StorageListResult {
```

Comment:
a reminder for default parameters for `list` and `getURL`

Reply:
https://github.com/aws-amplify/amplify-ios/issues/2223 for reminder

Reply:
This is the same as the functions which are being copied. I did not change it.

Reply:
Here is the existing API. I just dropped the listener parameter and changes this return type.

```swift
@discardableResult
func list(options: StorageListOperation.Request.Options?,
          resultListener: StorageListOperation.ResultListener?) -> StorageListOperation
```